jax/_src/numpy/lax_numpy.py

```diff
@@ -206,7 +207,7 @@ def _make_scalar_type(np_scalar_type: type) -> _ScalarMeta:
 issubsctype = dtypes.issubsctype
 promote_types = dtypes.promote_types
 
-ComplexWarning = np.ComplexWarning
+ComplexWarning = NumpyComplexWarning
```

Comment:
We'll need to clean up our own `jax.numpy` namespace to match the NEP, but we can do that in subsequent changes.